detections/application/splunk_command_and_scripting_interpreter_risky_spl_mltk.yml

```diff
@@ -0,0 +1,74 @@
+name: Splunk Command and Scripting Interpreter Risky SPL MLTK
+id: 19d0146c-2eae-4e53-8d39-1198a78fa9ca
+version: 1
+date: '2022-05-27'
+author: Abhinav Mishra, Kumar Sharad and Xiao Lin, Splunk
+type: Anomaly
+datamodel:
+- Splunk_Audit
+description: 'This detection utilizes machine learning model named "risky_command_abuse" trained from "Splunk Command 
+  and Scripting Interpreter Risky SPL MLTK Baseline". It should be scheduled to run hourly to detect whether a user 
+  has run searches containing risky SPL from this list 
+  https://docs.splunk.com/Documentation/Splunk/latest/Security/SPLsafeguards#Commands_that_trigger_the_warninga
+  with abnormally long running time in the past one hour, comparing with his/her past seven days history. This search 
+  uses the trained baseline to infer whether a search is an outlier (isOutlier ~= 1.0) or not (isOutlier~= 0.0)'
+search: '| tstats values(Search_Activity.user) AS user, sum(Search_Activity.total_run_time) AS run_time,
+  values(Search_Activity.search) as searches, count 
+  FROM datamodel=Splunk_Audit.Search_Activity WHERE (Search_Activity.user!="") 
+  AND (Search_Activity.total_run_time>1) AND (earliest=-1h@h latest=now) 
+  AND (Search_Activity.search IN ("*| runshellscript *", "*| collect *","*| delete *", "*| fit *", "*| outputcsv *", 
+  "*| outputlookup *", "*| run *", "*| script *", "*| sendalert *", "*| sendemail *", "*| tscollect *")) 
+  AND (Search_Activity.search_type=adhoc) AND (Search_Activity.user!=splunk-system-user) 
+  BY _time, Search_Activity.user span=1h 
+  | apply risky_command_abuse
+  | fields _time, user, searches, run_time, IsOutlier(run_time)
+  | rename IsOutlier(run_time) as isOutlier, _time as timestamp
```

Comment:
As written, since the algorithm is set to auto, the detection can create an anomaly if the run time is lower than normal as well. You can check other MLTK based detections as a guide to make sure that we are only looking at abnormally large.

Reply:
changed to use ower_threshold and upper_threshold, and set lower bound nearly zero.